predict.py

```diff
@@ -1,4 +1,9 @@
+import functools
```

Comment:
one predict to rule them all

Reply:
...which is, to be clear, great! unsure if that came through

Reply:
truly